pkg/ovsdb/transact.go

```diff
@@ -0,0 +1,1094 @@
+	}
+
+	value := condition[2]
+	return &Condition{Column: column, Function: fn, Value: value}, nil
+}
+
+func (c *Condition) CompareInteger(row *map[string]interface{}) (bool, error) {
+	actual, ok := (*row)[c.Column].(int)
+	if !ok {
+		return false, errors.New(E_CONSTRAINT_VIOLATION)
+	}
+	fn := c.Function
+	expected, ok := c.Value.(int)
+	if !ok {
+		return false, errors.New(E_CONSTRAINT_VIOLATION)
+	}
+	if (fn == FN_EQ || fn == FN_IN) && actual == expected {
+		return true, nil
+	}
+	if (fn == FN_NE || fn == FN_EX) && actual != expected {
+		return true, nil
+	}
+	if fn == FN_GT && actual > expected {
+		return true, nil
+	}
+	if fn == FN_GE && actual >= expected {
+		return true, nil
+	}
+	if fn == FN_LT && actual < expected {
+		return true, nil
+	}
+	if fn == FN_LE && actual <= expected {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (c *Condition) CompareReal(row *map[string]interface{}) (bool, error) {
+	actual, ok := (*row)[c.Column].(float64)
+	if !ok {
+		return false, errors.New(E_CONSTRAINT_VIOLATION)
+	}
+	fn := c.Function
+	expected, ok := c.Value.(float64)
+	if !ok {
+		return false, errors.New(E_CONSTRAINT_VIOLATION)
+	}
+
+	if (fn == FN_EQ || fn == FN_IN) && actual == expected {
+		return true, nil
+	}
+	if (fn == FN_NE || fn == FN_EX) && actual != expected {
+		return true, nil
+	}
+	if fn == FN_GT && actual > expected {
+		return true, nil
+	}
+	if fn == FN_GE && actual >= expected {
+		return true, nil
+	}
+	if fn == FN_LT && actual < expected {
+		return true, nil
+	}
+	if fn == FN_LE && actual <= expected {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (c *Condition) CompareBoolean(row *map[string]interface{}) (bool, error) {
+	actual, ok := (*row)[c.Column].(bool)
+	if !ok {
+		return false, errors.New(E_CONSTRAINT_VIOLATION)
+	}
+	fn := c.Function
+	expected, ok := c.Value.(bool)
+	if !ok {
+		return false, errors.New(E_CONSTRAINT_VIOLATION)
+	}
+
+	if (fn == FN_EQ || fn == FN_IN) && actual == expected {
+		return true, nil
+	}
+	if (fn == FN_NE || fn == FN_EX) && actual != expected {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (c *Condition) CompareString(row *map[string]interface{}) (bool, error) {
+	actual, ok := (*row)[c.Column].(string)
+	if !ok {
+		return false, errors.New(E_CONSTRAINT_VIOLATION)
+	}
+	fn := c.Function
+	expected, ok := c.Value.(string)
+	if !ok {
+		return false, errors.New(E_CONSTRAINT_VIOLATION)
+	}
+
+	if (fn == FN_EQ || fn == FN_IN) && actual == expected {
+		return true, nil
+	}
+	if (fn == FN_NE || fn == FN_EX) && actual != expected {
+		return true, nil
+	}
+	return false, nil
+}
+
+func (c *Condition) CompareUUID(row *map[string]interface{}) (bool, error) {
+	actual, ok := (*row)[c.Column].(libovsdb.UUID)
+	if !ok {
+		return false, errors.New(E_CONSTRAINT_VIOLATION)
+	}
+	fn := c.Function
+	expected, ok := c.Value.(libovsdb.UUID)
+	if !ok {
+		return false, errors.New(E_CONSTRAINT_VIOLATION)
+	}
+
+	if (fn == FN_EQ || fn == FN_IN) && actual == expected {
```

Comment:
UUID is a struct, I'm not sure that "==" will correctly work. Should we use deep copy, or their "GoUUID" values

Reply:
ok